templates/components/navigation-widget.html

```diff
@@ -14,6 +14,6 @@
 
 {% if URL_NAMESPACE %}
     {% sitetree_tree from URL_NAMESPACE template "sitetree/sidebar_menu_root.html" %}
-{% endif %}
-
+{% elif URL_NAME %}
     {% sitetree_tree from URL_NAME template "sitetree/sidebar_menu_root.html" %}
+{% endif %}
```

Comment:
In theory, this makes the 3 tests start to pass
https://app.travis-ci.com/github/python/pythondotorg/builds/235762782#L1343-L1648

I'm not sure this is the correct way to fix this.

Anyway, thanks @lucianoratamero for helping me debug this error.

Reply:
From what I could understand, [this context processor](https://github.com/python/pythondotorg/blob/main/pydotorg/context_processors.py#L9), for some reason, wasn't raising the `Resolver404` exception for db-routes such as is happening with tests related to `Page` or `Release` objects. 

I also checked the `PageView` implementation to confirm if it wasn't doing something unexpected but I didn't see anything out of place.

I tested 2 type of dynamic pages to check what would break if the path wasn't configured correctly. If I access an URL (`/lorem-ipsum/`) mapped to a `Page` object with the URL not being listed under the sitetree, I can see the content but without the navigation links as you can see here:

![Screenshot from 2021-08-24 12-03-50](https://user-images.githubusercontent.com/238223/130641490-b03bf090-daaf-4943-bdbc-a0fa8cccf05b.png)

If update this same object to have a path which is listed under the site tree, such as `/about/gettingstarted/`, the user sees both the page content and the navigation links as expected. Here's the print screen:

![Screenshot from 2021-08-24 12-04-27](https://user-images.githubusercontent.com/238223/130641794-16d51e4f-a50a-46bb-b8ea-3738e2d8c625.png)

So, I think this fix is a good one :+1: